source/puddlestuff/util.py

```diff
@@ -323,6 +324,10 @@ def write(audio, tags, save_mtime=True, justrename=False):
         audio.update(undo)
         audio.preview = preview
         raise
+    
+    except MutagenError as err:
+        audio.update(undo)
+        logging.exception(err)
```

Comment:
i'm wondering if we should also present a pop-up to the user?

Reply:
It seems essential, otherwise people might be confused as to why it's not saving. Either a message that informs them that the files are read-only or grey out the "Save" button and menu entry? Perhaps even both?

Reply:
The latest commit now causes an error dialog to appear if there are insufficient permissions to access the file.